### PR TITLE
move private_bytes to a shared method so we can reuse it

### PR DESCRIPTION
**Pull request by reaperhulk**
This will make things much simpler for EC and DSA.

### CODE REVIEW

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 1]
```diff
@@ -1093,6 +1093,65 @@ def _ec_key_set_public_key_affine_coordinates(self, ctx, x, y):
 
         return ctx
 
+    def _private_bytes(self, encoding, format, encryption_algorithm,
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py]: I think this method name should be "private_key_bytes", so we have context.

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 2]
```diff
@@ -1093,6 +1093,65 @@ def _ec_key_set_public_key_affine_coordinates(self, ctx, x, y):
...
+
+        # This is a temporary check until we land DER serialization.
+        if encoding is not serialization.Encoding.PEM:
+            raise ValueError("Only PEM encoding is supported by this backend")
+
+        if format is serialization.Format.PKCS8:
+            write_bio = self._lib.PEM_write_bio_PKCS8PrivateKey
+            key = evp_pkey
+        elif format is serialization.Format.TraditionalOpenSSL:
+            write_bio = pem_write_func
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py]: `pem_write_func` isn't a great name for what this is, it's not like PKCS8 isn't pem.

Comment (reaperhulk) [inline src/cryptography/hazmat/backends/openssl/backend.py]: Could be "traditional_write_func".

Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py]: Sounds good.

**Comment by alex** [comment]
lgtm

**Pull request merged**